src/main/java/net/fabricmc/filament/FilamentGradlePlugin.java

```diff
@@ -2,17 +2,27 @@
 
 import org.gradle.api.Plugin;
 import org.gradle.api.Project;
+import org.gradle.api.plugins.BasePlugin;
 
 import net.fabricmc.filament.task.CombineUnpickDefinitionsTask;
 import net.fabricmc.filament.task.GeneratePackageInfoMappingsTask;
-import net.fabricmc.filament.task.JavadocLintTask;
+import net.fabricmc.filament.task.MappingLintTask;
 import net.fabricmc.filament.task.RemapUnpickDefinitionsTask;
+import net.fabricmc.filament.task.lint.Checker;
 
 public final class FilamentGradlePlugin implements Plugin<Project> {
 	@Override
 	public void apply(Project project) {
 		project.getTasks().register("generatePackageInfoMappings", GeneratePackageInfoMappingsTask.class);
-		project.getTasks().register("javadocLint", JavadocLintTask.class);
+		var mappingLint = project.getTasks().register("mappingLint", MappingLintTask.class, task -> {
+			task.setGroup("verification");
+			task.setDescription("Checks that all mappings follow the conventions.");
+			task.getCheckers().addAll(Checker.DEFAULT_CHECKERS);
+		});
+
+		project.getPluginManager().withPlugin("org.gradle.base", p -> {
+			project.getTasks().named(BasePlugin.CLEAN_TASK_NAME, check -> check.dependsOn(mappingLint));
```

Comment:
Do you mean to refer to the check task than the clean task?

Reply:
Yep, I can't read apparently :eyes:

Reply:
Fixed.